lib/packageManagers.ts

```diff
@@ -1,6 +1,18 @@
 import { existsSync } from 'fs';
 
-type PackageManager = 'npm' | 'pnpm' | 'yarn';
+const supportedPackageManagers = ['npm', 'pnpm', 'yarn'] as const;
+type PackageManager = (typeof supportedPackageManagers)[number];
+
+const installCommands: Record<PackageManager, string> = {
+    npm: 'npm ci',
+    pnpm: 'pnpm install --frozen-lockfile',
+    yarn: 'yarn install --frozen-lockfile',
```

Comment:
We should update this. `yarn` no longer use `--frozen-lockfile`. the correct flag is now `--immutable`

Reply:
Another great pickup! Do we need to decide based on the Yarn version or are we happy to only support versions >= the version that introduced `--immutable`?

Reply:
It's there in Yarn Berry since version 2.0.0. I think it's safe for us to assume that no one ever run this pipeline in Yarn Classic. If they ever do it, probably they made another mistake and need to upgrade their package manager to Yarn Berry 😄

Reply:
Excellent - pushed a change to use `--immutable`

Reply:
Haha, [Use --immutable for Yarn and debug from environment](https://github.com/aligent/nx-serverless-deploy-pipe/pull/13/commits/9f6277b12b88d020e3d1de854213b46f870a4bce) didn't actually use `--immutable` 🤣 .